### PR TITLE
Update adding/removing cluster to/from cluster set rbac

### PR DESCRIPTION
**Pull request by showeimer**
APIs have a webhook that is checking for a different set of permissions than just simply patching labels:

https://github.com/open-cluster-management/registration/blob/08d8569c94bedfbe355e4d8a3930fce1bcccac8b/pkg/webhook/cluster/validating_webhook.go#L264-L269

### CODE REVIEW

File: frontend/src/lib/rbac-util.ts [hunk 1]
```diff
@@ -105,17 +105,27 @@ export async function checkAdminAccess() {
 }
 
 type Verb = 'get' | 'patch' | 'create' | 'delete' | 'update'
-
-export function rbacResource(verb: Verb, resource: IResource, namespace?: string, name?: string) {
+type SubResource = 'join'
```
Comment (hanqiuzh) [inline frontend/src/lib/rbac-util.ts:108]: we will only need join?

Comment (showeimer) [reply frontend/src/lib/rbac-util.ts:108]: Currently this is the only subresource I know of; and that we currently will use I guess.  Are there more?

Comment (hanqiuzh) [reply frontend/src/lib/rbac-util.ts:108]: no, just feel this code is a little bit confusing, but I'm good with that 👍

**Pull request merged**